### PR TITLE
fix(argo-cd): Create notifications clusterrole and clusterrolebinding when enabled

### PR DESCRIPTION
**Pull request by metajiji**
This PR created by following comment https://github.com/argoproj/argo-helm/pull/2315#issuecomment-1790049879

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Review by yu-croco — CHANGES_REQUESTED** [review]
Thank you for the PR. Please update `artifacthub.io/changes` block in Chart,yaml . 

> I have updated the chart [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog) with all the changes that come with this pull request according to changelog.

File: charts/argo-cd/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ appVersion: v2.8.6
 kubeVersion: ">=1.23.0-0"
 description: A Helm chart for Argo CD, a declarative, GitOps continuous delivery tool for Kubernetes.
 name: argo-cd
-version: 5.50.0
+version: 5.51.0
```
Comment (yu-croco) [inline charts/argo-cd/Chart.yaml:6]: I think it's `fixed` PR instead of `feat`, so Maybe bumping patch version is ok.

**Comment by metajiji** [comment]
> Thank you for the PR. Please update `artifacthub.io/changes` block in Chart,yaml .
> 
> > I have updated the chart [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog) with all the changes that come with this pull request according to changelog.

Done!

**Review by mbevc1 — APPROVED** [review]

File: charts/argo-cd/Chart.yaml (hunk 1 above)
Comment (yu-croco) [reply charts/argo-cd/Chart.yaml:6]: @metajiji 
What about this? 👀

**Review by mkilchhofer — APPROVED** [review]

File: charts/argo-cd/Chart.yaml (hunk 1 above)
Comment (metajiji) [reply charts/argo-cd/Chart.yaml]: Sorry, i agree yes this is patch. Done!

**Review by yu-croco — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**